save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt

```diff
@@ -68,6 +69,9 @@ class SaveAgent(private val config: AgentConfiguration,
     @Suppress("WHEN_WITHOUT_ELSE")  // when with sealed class
     private suspend fun startHeartbeats() = coroutineScope {
         println("Scheduling heartbeats")
+        runCatching {
+            saveAdditionalData()
```

Comment:
Maybe, we can create a new inheritor of `Heartbeat` class with additional data? So that all data is submitted in a unified way

Reply:
I think, we should split this information. `Heartbeat` pass to orchestrator only its status, not additional information.

Reply:
Ok, this makes sense. My concern was about logic that we have for backend queries: retires, etc. You could extract it into a separate method and reuse here